libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Recognizer.cs

```diff
@@ -0,0 +1,510 @@
+                response.EnsureSuccessStatusCode();
+                luisResponse = (JObject)JsonConvert.DeserializeObject(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+                var prediction = (JObject)luisResponse["prediction"];
+                recognizerResult = new RecognizerResult
+                {
+                    Text = utterance,
+                    AlteredText = prediction["alteredQuery"]?.Value<string>(),
+                    Intents = LuisV3Util.GetIntents(prediction),
+                    Entities = LuisV3Util.ExtractEntitiesAndMetadata(prediction),
+                };
+                LuisV3Util.AddProperties(prediction, recognizerResult);
+                if (_includeApiResults)
+                {
+                    recognizerResult.Properties.Add("luisResult", luisResponse);
+                }
+            }
+
+            // Log telemetry
+            await OnRecognizerResultAsync(recognizerResult, turnContext, telemetryProperties, telemetryMetrics, cancellationToken).ConfigureAwait(false);
+
+            var traceInfo = JObject.FromObject(
+                new
+                {
+                    recognizerResult,
+                    luisModel = new
+                    {
+                        ModelID = _application.ApplicationId,
+                    },
+                    luisOptions = options,
+                    luisResult = luisResponse,
+                });
+
+            await turnContext.TraceActivityAsync("LuisV3Recognizer", traceInfo, LuisTraceType, LuisTraceLabel, cancellationToken).ConfigureAwait(false);
+            return recognizerResult;
+        }
+
+        private LuisV3PredictionOptions MergeDefaultOptionsWithProvidedOptions(LuisV3PredictionOptions defaultOptions, LuisV3PredictionOptions overridenOptions)
+            => new LuisV3PredictionOptions()
+            {
+                DatetimeReference = overridenOptions.DatetimeReference ?? defaultOptions.DatetimeReference,
+                DynamicLists = overridenOptions.DynamicLists ?? defaultOptions.DynamicLists,
+                ExternalEntities = overridenOptions.ExternalEntities ?? defaultOptions.ExternalEntities,
+                IncludeAllIntents = overridenOptions.IncludeAllIntents ?? defaultOptions.IncludeAllIntents,
+                IncludeInstanceData = overridenOptions.IncludeInstanceData ?? defaultOptions.IncludeInstanceData,
+                Log = overridenOptions.Log ?? defaultOptions.Log,
+                PreferExternalEntities = overridenOptions.PreferExternalEntities ?? defaultOptions.PreferExternalEntities,
+                Slot = overridenOptions.Slot ?? defaultOptions.Slot,
+                Version = overridenOptions.Version ?? defaultOptions.Version,
+            };
+
+        private DelegatingHandler CreateHttpHandlerPipeline(HttpClientHandler httpClientHandler, params DelegatingHandler[] handlers)
+        {
```

Comment:
I'm not really clear on when/how/what this is doing. 

IS this sometime we still need?